exonum/src/runtime/mod.rs

```diff
@@ -251,17 +246,22 @@ pub trait Runtime: Send + Debug + 'static {
 
     /// Calls `before_commit` for all the services stored in the runtime.
```

Comment:
The core calls this method for each service instance started in this runtime, doesn't it?